ReadMe

Ubuntu Kylin Software Center

A new built software center client for Ubuntu Kylin. The aim is to improve localization experience. The applications and related data come from both ubuntu and ubuntu kylin package sources.


###############
Application management interfaces
open_cache
get_category_list
get_category_byname
get_category_apps
get_application_by_name
get_recommend_apps
get_ubuntukylin_apps
get_toprated_apps
get_review_rating_stats
get_application_reviews
get_application_screenshots



changelog
ubuntu-kylin-software-center (1.4.1) bionic; urgency=medium

  * fix up: max advertising button,update uksc,add icons,screenshop

 -- wukiage <wukaige@kylinos.cn>  Wed, 31 Jan 2018 17:33:11 +0800

ubuntu-kylin-software-center (1.4) artful; urgency=medium

  * update  UI interface 
  * fix up: install local software goto home interface
  * update uksc and add screenshots and icons 
  * adjustment screenshot display 

 -- wukaige <wukaige@kylinos.cn>  Fri, 22 Dec 2017 12:02:12 +0800

ubuntu-kylin-software-center (1.3.14) artful; urgency=medium

  *  update display page and main interface 
  *  more recommended software and update UKSC

 -- wukaige <wukaige@kylinos.cn>  Fri, 10 Nov 2017 17:23:54 +0800

ubuntu-kylin-software-center (1.3.13) artful; urgency=medium

  * fix up: update the software source for the first time when sources.list wrong,software cannot start
  * add settings interface,can update and edit sources.list    
  * software startup acceleration
  * software details interface edit
  * databases uksc add and translation

 -- wukaige <wukaige@kylinos.cn>  Wed, 06 Sep 2017 15:19:40 +0800

ubuntu-kylin-software-center (1.3.12) xenial; urgency=medium

  * fix:bug reinstall

 -- wukaige <wukaige@kylinos.cn>  Wed, 30 Aug 2017 16:24:21 +0800

ubuntu-kylin-software-center (1.3.11) xenial; urgency=medium

  *  improve and perfect

 -- wukaige <wukaige@kylinos.cn>  Mon, 21 Aug 2017 13:52:07 +0800

ubuntu-kylin-software-center (1.3.10) xenial; urgency=medium

  * fixed bug LP: #1559852,The UKSC freeze for a few seconds after it 
    installs or uninstalls app successfully.
  * fixed bug LP: #1561848,In allpage, the last categorybutton show 
    abnormally when mouse hover on it.
  * fixed bug LP: #1559955,While the downloading page opened, adjust the 
    scroll bar at the right side of uksc, the Downloading page display abnormal.
  * improve ubuntu-kylin-software-center self update.

 -- Shine <shenghuang@ubuntukylin.com>  Fri, 18 Mar 2016 14:21:25 +0800

ubuntu-kylin-software-center (1.3.9) xenial; urgency=medium

  * It supports the display and install, uninstall all the software
    source software.
  * Add a list of installed packages.
  * xapian support all software source inside search.
  * fixed bug LP: #1552647,Ubuntu Kylin Software Center has failed 
    to start. 
  * fixed bug LP: #1530337,The software sources update operation invalid.
  * fixed bug LP: #1538153,The prompt box for source update still appeared
    after the software sources update operation has been performed.

 -- Shine <shenghuang@ubuntukylin.com>  Fri, 18 Mar 2016 14:21:25 +0800

ubuntu-kylin-software-center (1.3.8) xenial; urgency=medium

  * fixed bug LP: #1550107,"Sudo apt-get update" command can not be 
    fully implemented.
  * LP: #1550099,Click on the close button in the prompt box for source 
    update is invalid.
  * LP: #1538152,After the system has started up for the first time, 
    the software sources update operation could not be loaded graphical 
    interface.
  * refresh page when update source by uksc.

 -- Shine <shenghuang@ubuntukylin.com>  Thu, 03 Mar 2016 14:22:44 +0800

ubuntu-kylin-software-center (1.3.7) xenial; urgency=medium

  * LP: #1535550 can't start uksc without giving password at uksc 
    first time start when update "source.list".
  * LP: #1535548 no update progress when first time start uksc and 
    update the "source.list".

 -- Shine <shenghuang@ubuntukylin.com>  Tue, 19 Jan 2016 17:14:20 +0800

ubuntu-kylin-software-center (1.3.6) wily; urgency=medium

  * To optimize the postinst.
  * fixed Bug #1483978 Bug #1484854.
  * improve the func which check the uksc working or not.

 -- Shine <shenghuang@ubuntukylin.com>  Mon, 14 Sep 2015 10:13:26 +0800

ubuntu-kylin-software-center (1.3.5) wily; urgency=medium

  * Fix postinst script

 -- Aron Xu <happyaron@ubuntu.com>  Thu, 30 Jul 2015 02:22:00 +0800

ubuntu-kylin-software-center (1.3.3) vivid; urgency=medium

  * Add the installation, upgrade, uninstall software progress 
    bar display.
  * Optimize the list, add the mouse hover effect.
  * Optimize the state of the progress bar, bugs when dealing 
    with broken network download and install.
  * Optimize the blurbs, add a layer of the gradient background, 
    solve the problem of dislocation window after expanding ads.
  * fix many bugs.

 -- Shine <shenghuang@ubuntukylin.com>  Tue, 21 Jul 2015 14:51:52 +0800

ubuntu-kylin-software-center (1.3.2) vivid; urgency=medium
  
  * New upstream bug fix release
    LP: #1372283, #1396402, #1420670, #1467722

 -- Shine <shenghuang@ubuntukylin.com>  Tue, 23 Jun 2015 18:07:24 +0800

ubuntu-kylin-software-center (1.3.1) vivid; urgency=medium

  * New upstream bug fix release
    LP: #1420644, #1420651, #1420636.

 -- Aron Xu <happyaron@ubuntu.com>  Tue, 24 Mar 2015 20:58:45 +0800

ubuntu-kylin-software-center (1.3.0) vivid; urgency=medium

  * Resources upload & download support
  * Sync App Icon from server
  * Add user translation interface in client

 -- Shine <shenghuang@ubuntukylin.com>  Wed, 11 Feb 2015 10:02:59 +0800

ubuntu-kylin-software-center (1.2.3.0) vivid; urgency=medium

  * LP: #1409187 : When enter the live mode, there will popup 
    a dialog about archive source update tip which is asking the 
    user to update the source or not.
  * LP: #1402953 : Part of the software starts up again after 
    closing will prompt "×× already running."

 -- Shine <shenghuang@ubuntukylin.com>  Tue, 20 Jan 2015 16:52:03 +0800

ubuntu-kylin-software-center (1.2.2.0) vivid; urgency=medium

  * Fixed the problem that interface can not receive the errorcode signal
  * Only upgradeable app,installed app can be searched in UPPAGE, UNPAGE
  * Improve the detail logic in ui (normalcard.py,detailscrollwidget.py 
    etc)
  * LP: #1396051, #1396403, #1396428, #1368208, #1379117, #1373916, #1396041, #1402898

 -- Shine <shenghuang@ubuntukylin.com>  Mon, 15 Dec 2014 14:21:04 +0800


ui/confw.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'confw.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_ConfigWidget(object):
    def setupUi(self, ConfigWidget):
        ConfigWidget.setObjectName(_fromUtf8("ConfigWidget"))
        ConfigWidget.resize(568, 480)
        self.pageListWidget = QtGui.QListWidget(ConfigWidget)
        self.pageListWidget.setGeometry(QtCore.QRect(16, 33, 96, 391))
        self.pageListWidget.setObjectName(_fromUtf8("pageListWidget"))
        self.sourceWidget = QtGui.QWidget(ConfigWidget)
        self.sourceWidget.setGeometry(QtCore.QRect(130, 50, 480, 361))
        self.sourceWidget.setObjectName(_fromUtf8("sourceWidget"))

        self.userWidget = QtGui.QWidget(ConfigWidget)
        self.userWidget.setGeometry(QtCore.QRect(0, 0, 480, 461))
        self.userWidget.setObjectName(_fromUtf8("userWidget"))
        self.groupBox_user = QtGui.QGroupBox(self.userWidget)
        self.groupBox_user.setGeometry(QtCore.QRect(10, 90, 472, 185))
	self.text2 = QtGui.QLabel(self.groupBox_user)
        self.text2.setGeometry(QtCore.QRect(140, 100, 100, 20))
        self.text2.setText(_fromUtf8(""))
        self.text2.setObjectName(_fromUtf8("text2"))

        self.text3 = QtGui.QLabel(self.groupBox_user)
        self.text3.setGeometry(QtCore.QRect(160, 130, 200, 20))
        self.text3.setText(_fromUtf8(""))
        self.text3.setObjectName(_fromUtf8("text3"))

        self.text4 = QtGui.QLabel(self.groupBox_user)
        self.text4.setGeometry(QtCore.QRect(160, 160, 300, 20))
        self.text4.setText(_fromUtf8(""))
        self.text4.setObjectName(_fromUtf8("text4"))

        self.text1 = QtGui.QLabel(self.sourceWidget)
        self.text1.setGeometry(QtCore.QRect(0, 100, 80, 17))
        self.text1.setText(_fromUtf8(""))
        self.text1.setObjectName(_fromUtf8("text1"))
        self.splitline = QtGui.QLabel(self.sourceWidget)
        self.splitline.setGeometry(QtCore.QRect(0, 20, 408, 1))
        self.splitline.setStyleSheet(_fromUtf8(""))
        self.splitline.setText(_fromUtf8(""))
        self.splitline.setObjectName(_fromUtf8("splitline"))
        self.groupBox = QtGui.QGroupBox(self.sourceWidget)
        self.groupBox.setGeometry(QtCore.QRect(0, 10, 411, 121))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(148, 144, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(148, 144, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(190, 190, 190))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.groupBox.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(9)
        self.groupBox.setFont(font)
        self.groupBox.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.groupBox.setObjectName(_fromUtf8("groupBox"))
        self.checkBox = QtGui.QCheckBox(self.groupBox)
        self.checkBox.setGeometry(QtCore.QRect(10, 60, 81, 24))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Text, brush)
        brush = QtGui.QBrush(QtGui.QColor(100, 100, 100))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.ToolTipText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Text, brush)
        brush = QtGui.QBrush(QtGui.QColor(100, 100, 100))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.ToolTipText, brush)
        brush = QtGui.QBrush(QtGui.QColor(190, 190, 190))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Text, brush)
        brush = QtGui.QBrush(QtGui.QColor(100, 100, 100))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.ToolTipText, brush)
        self.checkBox.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        font.setBold(False)
        font.setWeight(50)
        self.checkBox.setFont(font)
        self.checkBox.setChecked(False)
        self.checkBox.setObjectName(_fromUtf8("checkBox"))
        self.checkBox_2 = QtGui.QCheckBox(self.groupBox)
        self.checkBox_2.setGeometry(QtCore.QRect(10, 80, 81, 24))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        self.checkBox_2.setFont(font)
        self.checkBox_2.setObjectName(_fromUtf8("checkBox_2"))
        self.checkBox_3 = QtGui.QCheckBox(self.groupBox)
        self.checkBox_3.setGeometry(QtCore.QRect(90, 60, 81, 24))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        self.checkBox_3.setFont(font)
        self.checkBox_3.setObjectName(_fromUtf8("checkBox_3"))
        self.checkBox_4 = QtGui.QCheckBox(self.groupBox)
        self.checkBox_4.setGeometry(QtCore.QRect(150, 60, 91, 24))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        self.checkBox_4.setFont(font)
        self.checkBox_4.setObjectName(_fromUtf8("checkBox_4"))
        self.checkBox_5 = QtGui.QCheckBox(self.groupBox)
        self.checkBox_5.setGeometry(QtCore.QRect(240, 60, 81, 24))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        font.setBold(False)
        font.setWeight(50)
        self.checkBox_5.setFont(font)
        self.checkBox_5.setObjectName(_fromUtf8("checkBox_5"))
        self.checkBox_6 = QtGui.QCheckBox(self.groupBox)
        self.checkBox_6.setGeometry(QtCore.QRect(320, 60, 91, 24))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Serif"))
        font.setPointSize(10)
        font.setBold(False)
        font.setWeight(50)
        self.checkBox_6.setFont(font)
        self.checkBox_6.setObjectName(_fromUtf8("checkBox_6"))
        self.lesource = QtGui.QLineEdit(self.groupBox)
        self.lesource.setGeometry(QtCore.QRect(10, 30, 391, 20))
        self.lesource.setObjectName(_fromUtf8("lesource"))
        self.btnAdd = QtGui.QPushButton(self.groupBox)
        self.btnAdd.setGeometry(QtCore.QRect(310, 90, 90, 20))
        self.btnAdd.setText(_fromUtf8(""))
        self.btnAdd.setObjectName(_fromUtf8("btnAdd"))
        self.groupBox_2 = QtGui.QGroupBox(self.sourceWidget)
        self.groupBox_2.setGeometry(QtCore.QRect(0, 130, 411, 231))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(148, 144, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(148, 144, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(190, 190, 190))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.groupBox_2.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(9)
        self.groupBox_2.setFont(font)
        self.groupBox_2.setObjectName(_fromUtf8("groupBox_2"))
        self.btnUpdate = QtGui.QPushButton(self.groupBox_2)
        self.btnUpdate.setGeometry(QtCore.QRect(310, 200, 90, 20))
        self.btnUpdate.setText(_fromUtf8(""))
        self.btnUpdate.setObjectName(_fromUtf8("btnUpdate"))
        self.processwidget = QtGui.QWidget(self.groupBox_2)
        self.processwidget.setGeometry(QtCore.QRect(0, 190, 271, 31))
        self.processwidget.setObjectName(_fromUtf8("processwidget"))
        self.btnCancel = QtGui.QPushButton(self.processwidget)
        self.btnCancel.setGeometry(QtCore.QRect(250, 10, 13, 13))
        self.btnCancel.setText(_fromUtf8(""))
        self.btnCancel.setObjectName(_fromUtf8("btnCancel"))
        self.progressBar = QtGui.QProgressBar(self.processwidget)
        self.progressBar.setGeometry(QtCore.QRect(10, 10, 194, 17))
        self.progressBar.setProperty("value", 24)
        self.progressBar.setObjectName(_fromUtf8("progressBar"))
        self.sourceListWidget = QtGui.QListWidget(self.groupBox_2)
        self.sourceListWidget.setGeometry(QtCore.QRect(10, 30, 391, 161))
        self.sourceListWidget.setObjectName(_fromUtf8("sourceListWidget"))
        self.bg = QtGui.QLabel(ConfigWidget)
        self.bg.setGeometry(QtCore.QRect(0, 0, 568, 480))
        self.bg.setText(_fromUtf8(""))
        self.bg.setObjectName(_fromUtf8("bg"))
        self.btnClose = QtGui.QPushButton(ConfigWidget)
        self.btnClose.setGeometry(QtCore.QRect(130, 17, 28, 35))
        self.btnClose.setText(_fromUtf8(""))
        self.btnClose.setObjectName(_fromUtf8("btnClose"))
        self.label = QtGui.QLabel(ConfigWidget)
        self.label.setGeometry(QtCore.QRect(16, 32, 96, 1))
        self.label.setText(_fromUtf8(""))
        self.label.setObjectName(_fromUtf8("label"))
        self.cbhideubuntu = QtGui.QPushButton(ConfigWidget)
        self.cbhideubuntu.setGeometry(QtCore.QRect(190, 440, 100, 20))
        self.cbhideubuntu.setText(_fromUtf8(""))
        self.cbhideubuntu.setObjectName(_fromUtf8("cbhideubuntu"))
        self.btnReset = QtGui.QPushButton(ConfigWidget)
        self.btnReset.setGeometry(QtCore.QRect(330, 430, 90, 20))
        self.btnReset.setText(_fromUtf8(""))
        self.btnReset.setObjectName(_fromUtf8("btnReset"))
        self.bg.raise_()
        self.pageListWidget.raise_()
        self.sourceWidget.raise_()
        self.btnClose.raise_()
        self.label.raise_()
        self.cbhideubuntu.raise_()
        self.btnReset.raise_()

        self.retranslateUi(ConfigWidget)
        QtCore.QMetaObject.connectSlotsByName(ConfigWidget)

    def retranslateUi(self, ConfigWidget):
        ConfigWidget.setWindowTitle(_translate("ConfigWidget", "Form", None))
        self.groupBox.setTitle(_translate("ConfigWidget", "软件源配置", None))
        self.checkBox.setText(_translate("ConfigWidget", "deb", None))
        self.checkBox_2.setText(_translate("ConfigWidget", "deb-src", None))
        self.checkBox_3.setText(_translate("ConfigWidget", "main", None))
        self.checkBox_4.setText(_translate("ConfigWidget", "restricted", None))
        self.checkBox_5.setText(_translate("ConfigWidget", "universe", None))
        self.checkBox_6.setText(_translate("ConfigWidget", "multiverse", None))
        self.groupBox_2.setTitle(_translate("ConfigWidget", "软件源列表", None))


if __name__ == "__main__":
    import sys
    app = QtGui.QApplication(sys.argv)
    ConfigWidget = QtGui.QWidget()
    ui = Ui_ConfigWidget()
    ui.setupUi(ConfigWidget)
    ConfigWidget.show()
    sys.exit(app.exec_())

